### PR TITLE
Add debugging instructions

### PR DESCRIPTION
**Pull request by zaaath**
This adds instructions on how to set up debugging environment in VS Code for both, the engine and the sql client. Debugging specific tests works great as well.

Context: when I first started working my way through the toydb codebase, it was hard to understand what's going on (even though the docs, the tests, and the code itself are beautifully written!) without a debugger, and I couldn't find instructions on how to debug the code in the repo itself. I ended up googling around and setting up debugging environments on two laptops. With this PR I want to make it easier for the new engineers to onboard.

### CODE REVIEW

**Comment by leotoff** [comment]
@erikgrinaker is there something missing in this PR? what is the next step to get this approved?

**Comment by erikgrinaker** [comment]
Hey @leotoff! Thanks for the PR, and sorry for the late reply -- it's been a busy week. I'll get back to you in a bit.

File: README.md [hunk 1]
```diff
@@ -73,6 +73,19 @@ desirable but not yet implemented.
 Execute `cargo test` to run all tests, or check out the latest
 [CI run](https://cloud.drone.io/erikgrinaker/toydb).
 
+## Debugging
+
+VS Code provides a very intuitive environment for debugging toyDB. Follow these steps to set it up:
```
Comment (erikgrinaker) [inline README.md:78]:
```suggestion
[VSCode](https://code.visualstudio.com) provides a very intuitive environment for debugging toyDB. Follow these steps to set it up:
```

File: README.md [hunk 2]
```diff
@@ -73,6 +73,19 @@ desirable but not yet implemented.
...
+1. Install [CodeLLDB](https://marketplace.visualstudio.com/items?itemName=vadimcn.vscode-lldb) extension
```
Comment (erikgrinaker) [inline README.md:79]:
```suggestion
1. Install the [CodeLLDB](https://marketplace.visualstudio.com/items?itemName=vadimcn.vscode-lldb) extension
```

File: README.md [hunk 3]
```diff
@@ -73,6 +73,19 @@ desirable but not yet implemented.
...
+1. Go to "Run and Debug" tab and initialize launch.json, here is my autogenerated [`launch.json`](https://gist.github.com/leotoff/78e20cb1948e8bb20631271576730865) for reference
+
+The above lets debug `toysql` (from "Run and Debug" tab) and unit tests (using a popup in test files). Debugging the engine is a bit more involved:
+1. Add config arg `"-c=${workspaceFolder}/config/toydb.yaml"` to "Debug executable 'toydb'" configuration in `launch.json`, see [my configuration](https://gist.github.com/leotoff/b940df1b947b7d4e6fd42bd057943fe3) for reference
+1. Create a data directory `sudo mkdir /var/lib/toydb && sudo chown $(whoami) /var/lib/toydb` (the path is specified in `config/toydb.yaml`)
```
Comment (erikgrinaker) [inline README.md:84]: Maybe we should just change the default to `./data`. `/var/lib/toydb` is rarely useful.

File: README.md [hunk 4]
```diff
@@ -73,6 +73,19 @@ desirable but not yet implemented.
...
+1. Go to "Run and Debug" tab, select "Debug executable 'toydb'", run
+
+If the last command fails with `Address already in use (os error 48)`, go to Activity Monitor and send the "Quit" signal to `toydb` processes. There might be some lingering processes after running the cluster command.
```
Comment (erikgrinaker) [inline README.md:87]: This is specific to macOS, consider generalizing this for other OSes -- or at least Linux.

**Review by erikgrinaker — COMMENTED** [review]
Thanks for the PR, and for your patience! Added some suggestions, otherwise this looks good.

**Comment by erikgrinaker** [comment]
PS: please wrap lines at 100 chars, for consistency.

File: README.md (hunk 3 above)
Comment (leotoff) [reply README.md:84]: sounds good. do you want me to modify `config/toydb.yaml` in this or a separate PR? if in a separate PR, I'll modify README.md there as well (`/var/lib/toydb` -> `./data`). Let me know how to approach this.

File: README.md (hunk 4 above)
Comment (leotoff) [reply README.md:87]: Oops, that's a good callout, I'm on it.

**Comment by leotoff** [comment]
Sorry about the delay, I'm working on addressing outstanding issues.

Tasks:

- [ ]  Get directions and then execute them for `/var/lib/toydb` -> `./data` change in `config/toydb.yaml`
- [ ]  Generalize the process clean up step (Activity Monitor shouldn't be used)
- [ ]  Wrap lines at 100 chars

File: README.md (hunk 4 above)
Comment (leotoff) [reply README.md]: Resolved, I removed MacOS-specific tools.

File: README.md (hunk 3 above)
Comment (erikgrinaker) [reply README.md]: Separate PR would be great, thanks!

**Comment by erikgrinaker** [comment]
Whoops, totally forgot about this. Thanks again!

**Pull request merged**